engine/runtime/src/main/java/org/enso/interpreter/runtime/ThreadExecutors.java

```diff
@@ -40,7 +40,8 @@ ExecutorService newCachedThreadPool(
   }
 
   ScheduledExecutorService newScheduledThreadPool(int cnt, String name, boolean systemThread) {
-    var s = Executors.newScheduledThreadPool(cnt, new Factory(name, systemThread));
+    var s = new ScheduledThreadPoolExecutor(cnt, new Factory(name, systemThread));
+    s.allowCoreThreadTimeOut(true);
```

Comment:
- Is this [allowCoreThreadTimeOut](https://docs.oracle.com/en/java/javase/24/docs/api/java.base/java/util/concurrent/ThreadPoolExecutor.html#allowCoreThreadTimeOut(boolean)) necessary?
- shouldn't the `TruffleSafepoint.poll` be enough?

Reply:
It is necessary although it is not the same problem as for `EnsoPolyglotJava`. Core threads that are parked have no logic (or possibility to even extend) that ensures regular `TruffleSafepoint.poll` calls. My understanding is that Executors simply aren't Truffle-friendly since 24.x

Reply:
> My understanding is that Executors simply aren't Truffle-friendly since 24.x

- Yeah, the default executor implementations know nothing about Truffle.
- we'd be better if we [implemented own executor](https://github.com/enso-org/enso/pull/13907#issuecomment-3251793275)

Reply:
It is great we [have a test](https://github.com/enso-org/enso/pull/13907/files#diff-7b5331e76775d2b9cf961c00e5712485da50c5783166aed634c4f643ad372cbdR212) to simulate the deadlock. Alas, removing
```
s.allowCoreThreadTimeOut(true);
```
doesn't cause any deadlock when running 
```
sbt:enso> runtime-integration-tests/testOnly *RuntimeManagementTest -- -z multithreaded
```